src/libraries/Common/tests/TestUtilities/System/IO/FileCleanupTestBase.cs

```diff
@@ -208,10 +208,14 @@ private unsafe string GetTestDirectoryActualCasing()
                     // Remove extended prefix
                     int skip = PathInternal.IsExtended(buffer) ? 4 : 0;
 
-                    return new string(
+                    string str = new string(
                         buffer,
                         skip,
                         (int)result - skip);
+
+                    ArrayPool<char>.Shared.Return(buffer);
```

Comment:
This is test code. Why is it even using the ArrayPool?  Rather than fixing this by returning the buffer, I'd rather see L205 changed to just be `new char[4096]`.

There are other oddities here.  Why is `buffer` marked as nullable when it's never null?  And why is there an "Initial" buffer size when no other buffer size is ever used (eg there's no growing)?